.github/workflows/rust.yml

```
@@ -235,3 +235,26 @@ jobs:
       - name: Check the building of docs
         working-directory: ./rust
         run: cargo +nightly doc --all-features --color always
+
+  coverage:
+    name: cargo-tarpaulin
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repository
+        uses: actions/checkout@v2
+
+      - name: Install stable toolchain
+        uses: actions-rs/toolchain@v1
+        with:
+          toolchain: stable
+          override: true
+
+      - name: Run cargo-tarpaulin
+        uses: actions-rs/tarpaulin@v0.1.2
+        with:
+          args: '--manifest-path rust/Cargo.toml --all-features --force-clean --lib --exclude-files */tests/* src/vendor/* *.h'
```

Why do we need to specify the `*.h` files? Also I _think_ that to correctly ignore the test code we need `--ignore-tests`. 
```suggestion
          args: '--manifest-path rust/Cargo.toml --all-features --force-clean --lib --ignore-tests --exclude-files src/vendor/*'
```

Not 100% sure though